src/Main.hs

```diff
@@ -77,9 +74,8 @@ runQueryMode qattr inputF =
 
 readJournal :: FilePath -> IO [SimpleTransaction]
 readJournal journalPath = do
-  journalT <- TIO.readFile journalPath
-  journal' <- readJournal' journalT
-  return $ simpleTransactionsInJournal journal'
+  xml <- LP.xmlOf journalPath
+  return $ LX.simpleTransactionsInXmlDocument xml
```

Comment:
I think a better design would be changing from one `readJournal` (which uses hledger) to another 'readJournal' function.